llvm/lib/Target/X86/X86ISelDAGToDAG.cpp

```diff
@@ -2944,6 +2944,18 @@ bool X86DAGToDAGISel::selectMOV64Imm32(SDValue N, SDValue &Imm) {
     return false;
 
   Imm = N;
+  if (Subtarget->isTargetWindowsCygwin() && Subtarget->isTargetWin64()) {
```

Comment:
This seems to largely duplicate the generic code below. The only difference is when we've failed to get absolute symbol range for the GV. So, certainly there should not be any code duplication and test / example is required.